### ISSUE TITLE
Did you use data augment?

### ISSUE THREAD

**Issue by billhhh** [arnoweng/CheXNet#2]
Thx for sharing the code! did you use data augment?
and this seems only testing code, would you mind share training code also

Thanks

**Comment by arnoweng** [comment]
I've tried several data augmentation methods that work well in other domains, but they do not work in this task somehow. The training code is adopted from PyTorch examples and you can easily find them in offical page.

**Comment by jkhhh** [comment]
I ran your trained model on the testset, but it seems that something has been done to adjust the mean probability of each disease. Such values will not be obtained by the original training strategy (If my experiment is correct). 

![loss](https://user-images.githubusercontent.com/35034251/34487502-8fe5d06c-f00f-11e7-88ba-b3ea0e8dcf16.png)

**Comment by billhhh** [comment]
If without data augment, how to address data positive & negtive uneven problem? I think the original paper use data augment...

**Comment by ZhichengHuang** [comment]
what method do you use to address the class unbalanced?

**Comment by arnoweng** [comment]
The problem can be addressed by oversampling positive classes or changing sample weights which are elaborated in original paper [v1](https://arxiv.org/abs/1711.05225v1) and [v2](https://arxiv.org/abs/1711.05225v2) respectively.

**Comment by billhhh** [comment]
I do not think the v2 changed the weights, the paper says
![image](https://user-images.githubusercontent.com/7709725/34599575-1211a384-f22e-11e7-837a-f71bac471b73.png), and v1 says "We also augment the training
data with random horizontal flipping."

**Comment by billhhh** [comment]
Hello, so which way do you use? upsampling or changing sample weights?

**Comment by billhhh** [comment]
Sorry, I did not realize you have metioned "the up sampling or change weight" is about pneumonia binary classification problem. what I was asking is all about 14 classes classification problem

**Comment by chaoyan1037** [comment]
When training the model, did you freeze the model parameters except for the modified parts? Or, just train all parameters? @arnoweng

**Comment by arnoweng** [comment]
@chaoyan1073 I have tried both. This released model was trained without freezing parameters.

**Comment by chaoyan1037** [comment]
@arnoweng  Thanks for your kind reply! I have tried both, too. However, In my case, freezing partial parameters will produce better AUC score 0.810. But it is still not as good as your 0.847. But I did not adopt any sampling strategies at present. I will try some sampling skills and see if them helps.

**Comment by zoogzog** [comment]
Thank you @arnoweng for sharing the code. I have implemented the [training procedure](https://github.com/zoogzog/chexnet), and strangely enough, was able to obtain better AUROC score (0.8508).  Following imagenet example I used random crops and flips during training stage, learning rate was set to 0.0001.
